astropy/units/tests/test_quantity.py

```diff
@@ -468,6 +468,11 @@ def test_power(self):
         assert_array_almost_equal(new_quantity.value, 1489.355288, decimal=7)
         assert new_quantity.unit == u.Unit("m^3")
 
+    def test_quantity_as_power(self):
+        # raise unit to a dimensionless Quantity power
+        # regression test for https://github.com/astropy/astropy/issues/16260
+        assert u.m ** u.Quantity(2) == u.m**2
```

Comment:
Another case where this fails is with numpy integers:

```
In [4]: maybe_simple_fraction(np.int64(5))
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
Input In [4], in <cell line: 1>()
----> 1 maybe_simple_fraction(np.int64(5))

File ~/Projects/astropy/astropy/units/utils.py:217, in maybe_simple_fraction(p, max_denominator)
    215 if p == 0 or p.__class__ is int or p.__class__ is Fraction:
    216     return p
--> 217 n, d = p.as_integer_ratio()
    218 a = n // d
    219 # Normally, start with 0,1 and 1,0; here we have applied first iteration.

AttributeError: 'numpy.int64' object has no attribute 'as_integer_ratio'
```

Reply:
```
@pytest.mark.parametrize("exponent", [u.Quantity(2), np.uint64(2), np.int32(2), 2.0, np.float32(2.0)])
def test_unit_powers(exponent):
    assert u.m**exponent = u.m**2
```

Reply:
good point, I've merged my test with the previous one and parametrized it to better cover regressions